### PR TITLE
Fix is for XSD having the decimal data type with restriction

### PR DESCRIPTION
**Pull request by sivaponting**
This code fix is for XSD having the decimal data type with restriction. XSDToSchema util class returns the StringType if the XSD has the decimal data type with restriction.

### CODE REVIEW

File: src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala [hunk 1]
```diff
@@ -86,7 +86,10 @@ object XSDToSchema {
       case simpleType: XmlSchemaSimpleType =>
         val schemaType = simpleType.getContent match {
           case restriction: XmlSchemaSimpleTypeRestriction =>
-            simpleType.getQName match {
+            val qName = if (simpleType.getQName !=null) {
```
Comment (srowen) [inline src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala:89]: Just add a `case null =>` to the match below instead

File: src/test/scala/com/databricks/spark/xml/util/XSDToSchemaSuite.scala [hunk 2]
```diff
@@ -17,10 +17,8 @@
 package com.databricks.spark.xml.util
 
 import java.nio.file.Paths
-
-import org.apache.spark.sql.types.{ArrayType, FloatType, LongType, StringType}
+import org.apache.spark.sql.types.{ArrayType, DecimalType, FloatType, LongType, StringType}
```
Comment (srowen) [inline src/test/scala/com/databricks/spark/xml/util/XSDToSchemaSuite.scala:20]: Leave the newline spacing here as-is

File: src/test/scala/com/databricks/spark/xml/util/XSDToSchemaSuite.scala [hunk 3]
```diff
@@ -118,4 +116,11 @@ class XSDToSchemaSuite extends AnyFunSuite {
     assert(expectedSchema === parsedSchema)
   }
 
+  test("decimal-with-restriction") {
```
Comment (srowen) [inline src/test/scala/com/databricks/spark/xml/util/XSDToSchemaSuite.scala:119]: Maybe a slightly more descriptive name, but no big deal

**Review by srowen — CHANGES_REQUESTED** [review]
This needs to be open vs the master branch but looks OK as a change, modulo comments here. A slightly more descriptive title and description would be good

File: src/test/scala/com/databricks/spark/xml/util/XSDToSchemaSuite.scala (hunk 3 above)
Comment (sivaponting) [reply src/test/scala/com/databricks/spark/xml/util/XSDToSchemaSuite.scala]: Updated the test case name. Hope it's more descriptive

**Review by srowen — CHANGES_REQUESTED** [review]
Just update the title too

File: src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala [hunk 4]
```diff
@@ -86,7 +86,11 @@ object XSDToSchema {
       case simpleType: XmlSchemaSimpleType =>
         val schemaType = simpleType.getContent match {
           case restriction: XmlSchemaSimpleTypeRestriction =>
-            simpleType.getQName match {
+            val qName = simpleType.getQName match {
+              case null => restriction.getBaseTypeName
+              case _ => simpleType.getQName
```
[the remark this answers is not in the document]
Comment (srowen) [reply src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala:90]: No - the match statement _below_ this.

[the remark this answers is not in the document]
Comment (sivaponting) [reply src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala:90]: Hi srowen, the below match statement in returning the spark datatype. but the 
case null => restriction.getBaseTypeName returning QName.


The below snippet of code is added to find the QName
val qName = simpleType.getQName match {
              case null => restriction.getBaseTypeName
              case _ => simpleType.getQName
}

[the remark this answers is not in the document]
Comment (srowen) [reply src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala:90]: Ah I am misreading the code, yeah, sorry. OK do this:
```
val qName = simpleType.getQName match {
  case null => restriction.getBaseTypeName
  case n => n
}
```

[the remark this answers is not in the document]
Comment (srowen) [reply src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala:90]: `Option(simpleType.getQName).getOrElse(restriction.getBaseTypeName)` would be fine too

[the remark this answers is not in the document]
Comment (sivaponting) [reply src/main/scala/com/databricks/spark/xml/util/XSDToSchema.scala]: Hi srowen, based on ur suggestion updated the code. Pls review it. Thanks

**Review by srowen — APPROVED** [review]

**Pull request merged**